pkg/kv/bulk/sst_batcher.go

```diff
@@ -316,6 +316,11 @@ func (b *SSTBatcher) flushIfNeeded(ctx context.Context, nextKey roachpb.Key) err
 	}
 
 	if b.sstWriter.DataSize >= ingestFileSize(b.settings) {
+		prevRow, prevErr := keys.EnsureSafeSplitKey(b.batchEndKey)
+		nextRow, nextErr := keys.EnsureSafeSplitKey(nextKey)
+		if prevErr == nil && nextErr == nil && bytes.Equal(prevRow, nextRow) {
```

Comment:
Note to self: It looks like EnsureSafeSplitKey returns errors in the case that the key isn't a table key or is somehow malformed. Given that, it makes sense to ignore the non-nil errors here since both cases would likely fail somewhere else in this process.

Reply:
added some comments